packages/kernel-5.10/config-bottlerocket

```diff
@@ -40,6 +40,15 @@ CONFIG_IXGBE_DCB=y
 CONFIG_IXGBE_HWMON=y
 CONFIG_IXGBEVF=m
 
+# Mellanox network support
+CONFIG_MLXFW=m
+CONFIG_MLX5_CORE=m
+CONFIG_MLX5_INFINIBAND=m
+CONFIG_NET_VENDOR_MELLANOX=y
+CONFIG_MLX5_CORE_EN=y
+CONFIG_NET_SWITCHDEV=y
+CONFIG_MLX5_BRIDGE=n
```

Comment:
Does that mean it's safe to assume it defaults to `=n` and we wouldn't need to specify it here?

Reply:
In a way, yes.

The option does not exist in kernels prior to 5.14. It will be ignored as there is no code that switches anything off or on, no matter what we choose here. It will probably produce something like an unknown option log line somewhere, that is all.

No occurrence of `MLX5_BRIDGE` in the kernel for 5.4.124:

```
HEAD is now at 4f3fee72a74c Linux 5.10.124
[foersleo@devbox linux]$ grep -nr MLX5_BRIDGE *
[foersleo@devbox linux]$ 
```

On kernels after 5.14 however, it will default to `=y`:

```
config MLX5_BRIDGE
	bool
	depends on MLX5_ESWITCH && BRIDGE
	default y
	help
	  mlx5 ConnectX offloads support for Ethernet Bridging (BRIDGE).
	  Enable adding representors of mlx5 uplink and VF ports to Bridge and
	  offloading rules for traffic between such ports. Supports VLANs (trunk and
	  access modes).
```

There is some handy webpages, that can show some info on which kernel versions have certain options, without having to go through the versions oneself. So far I have not seen any problems with regards to reliability of the data (probably automagically created). For example https://cateee.net/lkddb/web-lkddb/MLX5_BRIDGE.html